omod-2.7/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_7/ConceptReferenceRangeResource2_7.java

```diff
@@ -0,0 +1,199 @@
+			description.addProperty("lowAbsolute");
+			description.addProperty("lowCritical");
+			description.addProperty("units");
+			description.addProperty("precise");
+			description.addSelfLink();
+			return description;
+		}
+		else if (rep instanceof RefRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			return description;
+		}
+		
+		return null;
+	}
+	
+	@PropertyGetter("units")
+	public String getUnits(ConceptReferenceRange instance) {
+		return instance.getConceptNumeric().getUnits();
+	}
+	
+	@PropertyGetter("precise")
+	public Boolean getPrecise(ConceptReferenceRange instance) {
+		return instance.getConceptNumeric().getAllowDecimal();
+	}
+	
+	@PropertyGetter("concept")
+	public String getConcept(ConceptReferenceRange instance) {
+		return instance.getConceptNumeric().getUuid();
+	}
+	
+	@PropertyGetter("display")
+	public String getDisplayString(ConceptReferenceRange instance) {
+		String localization = getLocalization("Concept", instance.getConceptNumeric().getUuid());
+		if (StringUtils.isNotBlank(localization)) {
+			return localization;
+		} else {
+			ConceptName cn = instance.getConceptNumeric().getName();
+			if (cn != null) {
+				return cn.getName();
+			} else {
+				return instance.toString();
+			}
+		}
+	}
+
+	@Override
+	public ConceptReferenceRange getByUniqueId(String uniqueId) {
+		return Context.getConceptService().getConceptReferenceRangeByUuid(uniqueId);
+	}
+
+	@Override
+	protected void delete(ConceptReferenceRange delegate, String reason, RequestContext context) throws ResponseException {
+		throw new UnsupportedOperationException("ConceptReferenceRange does not support this operation");
+	}
+
+	@Override
+	public void purge(ConceptReferenceRange delegate, RequestContext context) throws ResponseException {
+		Context.getConceptService().purgeConceptReferenceRange(delegate);
+	}
+
+	@Override
+	protected PageableResult doSearch(RequestContext context) {
+
+		ConceptService conceptService = Context.getConceptService();
+		List<ConceptReferenceRange> referenceRanges = new ArrayList<ConceptReferenceRange>();
+		
+		String patientUuid = context.getParameter("patient");
+		if (StringUtils.isBlank(patientUuid)) {
+			throw new IllegalArgumentException("patient is required");
+		}
+		Patient patient = Context.getPatientService().getPatientByUuid(patientUuid);
+		if (patient == null) {
+			throw new ObjectNotFoundException("No patient with uuid: " + patientUuid);
+		}
+
+		String conceptUuid = context.getParameter("concept");
+		if (StringUtils.isBlank(conceptUuid)) {
+			throw new IllegalArgumentException("concept is required");
+		}
+		
+		String[] conceptReferenceStrings = conceptUuid.split(",");
+		for (String conceptReference : conceptReferenceStrings) {
```

Comment:
Unimportant, but just to point out that in 2.6+, we actually have a method to do this in the standard `ConceptService `called `getConceptByReference()`, we just don't use it other places for backwards compatibility.

Reply:
I wonder how i completely forgot about this! It makes lots of sense and greatly simplifies the code. Addressed it here: https://github.com/openmrs/openmrs-module-webservices.rest/pull/652